presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -164,8 +192,34 @@ public NodeSelector createNodeSelector(Session session, ConnectorId connectorId,
                     networkLocationSegmentNames,
                     networkLocationCache);
         }
+        else if (resourceAwareSchedulingStrategy == TTL) {
```

Comment:
I am wondering if we should also add another condition for checking availability of execution estimates in session.

Reply:
I would recommend first creating the `SimpleNodeSelector`, then checking if the strategy is `TTL`, if so decorate that selector with a `SimpleTTLNodeSelector`, otherwise return it directly.